### PR TITLE
Reportback content

### LINKED ISSUE #162: Include a user's actual reportback content in GET /users/campaign responses
For responses that return user reportback data, instead of just including the reportback id, also respond with the actual reportback content.

```
GET /users/:term/:identifier/campaigns
and
GET /user/campaigns/:campaign_id
```

You can retrieve the rest of the user data by querying the Drupal API at:

```
GET /reportbacks/:reportback_id
```
https://github.com/DoSomething/dosomething/wiki/API#retrieve-a-specific-reportback

### PR DESCRIPTION
**Pull request by chloealee**
Fixes #162 

Edit to return all reportback content, beyond only reportback id.

### CODE REVIEW

File: app/Http/Controllers/CampaignController.php [hunk 1]
```diff
@@ -43,7 +43,16 @@ public function index($term, $id)
...
 
         $campaigns = $user->campaigns;
-        return $this->respond($campaigns);
+
+        foreach ($campaigns as $campaign) {
+            if ($campaign->reportback_id) {
+                $response = $this->drupal->reportbackContent($campaign->reportback_id);
+                $campaign['reportback_data'] = $response;
+            } else {
+                return $this->respond($campaigns);
```
Comment (angaither) [inline app/Http/Controllers/CampaignController.php]: does this `return $this->respond($campaigns);` need to be in an else, when you're doing it anyways in the line below?

File: app/Http/Controllers/CampaignController.php [hunk 2]
```diff
@@ -65,7 +74,13 @@ public function show($campaign_id)
             throw new NotFoundHttpException('User has not signed up for this campaign.');
         }
 
-        return $this->respond($campaign);
+        if ($campaign->reportback_id) {
+            $response = $this->drupal->reportbackContent($campaign->reportback_id);
+            $campaign['reportback_data'] = $response;
+            return $this->respond($campaign);
```
Comment (angaither) [inline app/Http/Controllers/CampaignController.php]: same

File: app/Http/Controllers/CampaignController.php (hunk 1 above)
Comment (chloealee) [inline app/Http/Controllers/CampaignController.php]: oops - that is redundant! I'll get rid of this and the one in the block below as well. Thanks!

**Comment by jonuy** [comment]
@chloealee which endpoints did you test against and can you copy and paste an example response in here? (just remember to strike out any sensitive data)

**Comment by chloealee** [comment]
@jonuy sorry about that! Something got messed up when I tried to rebase. Just fixed it and should work now. 

Here's the endpoint I'm testing against:
`http://northstar.dev:8000/v1/users/mobile/{mobile_number}/campaigns`

Here's an example response: 
{
  "data": [
    {
      "drupal_id": "***",
      "signup_id": *,
      "signup_source": "**",
      "updated_at": "2015-07-01 02:42:25",
      "created_at": "2015-06-30 19:31:50",
      "reportback_id": *,
      "reportback_data": {
        "data": {
          "id": "*",
          "created_at": "1394231347",
          "updated_at": "1394231347",
          "quantity": 12,
          "reportback_items": {
            "total": 1,
            "data": [
              {
                "id": "**",
                "caption": "DoSomething? Just did!",
                "uri": "http://staging.beta.dosomething.org/api/v1/reportback-items/41",
                "media": {
                  "uri": "http://staging.beta.dosomething.org/sites/default/files/styles/480x480/public/kittycat_0.jpg?itok=pI3zyrHb",
                  "type": "image"
                },
                "created_at": "1402428343",
                "kudos": null
              }
            ]
          },
          "campaign": {
            "id": "**",
            "title": "Party Animal",
            "reportback_info": {
              "copy": "Let us know how many items you included on your list! (Note: the shelter does NOT need an Xbox One.)",
              "confirmation_message": "Thanks for helping homeless animals. You deserve a party!\\\r\\\n",
              "noun": "items",
              "verb": "wished for"
            }
          },
          "user": {
            "id": "**"
          }
        }
      }
    }
  ]
}

**Comment by jonuy** [comment]
@chloealee could we get rid of that last nested `data` object in `reportback_data`? So instead of it being `obj.data[0].reportback_data.data` to get at the reportback data, it's just `obj.data[0].reportback_data`?

**Comment by jonuy** [comment]
Also, it seems like `GET /user/campaigns/:campaign_id` should also work since you're updating the `show()` function. I just wanna make sure that one's getting tested too.

**Comment by chloealee** [comment]
@jonuy yup - `GET /user/campaigns/:campaign_id` is working too! Just has the same issue as the `index()` with the extra nested `data` object in `reportback_data` - figuring out how to remove now - will keep you updated!

**Comment by chloealee** [comment]
@jonuy removed the nested `data` object in `reportback_data` - should be structured correctly now!

File: app/Http/Controllers/CampaignController.php [hunk 3]
```diff
@@ -60,12 +68,18 @@ public function show($campaign_id)
         $user = User::current();
 
         $campaign = $user->campaigns()->where('drupal_id', $campaign_id)->first();
-
+        echo("hello");
```
Comment (jonuy) [inline app/Http/Controllers/CampaignController.php]: :no_good:

File: app/Services/DrupalAPI.php [hunk 4]
```diff
@@ -251,4 +251,12 @@ public function storeKudos($drupal_id, $request)
 
         return $body;
     }
+
```
Comment (jonuy) [inline app/Services/DrupalAPI.php]: Could you drop in a comment block for the new `reportbackContent()` function too?

File: app/Http/Controllers/CampaignController.php (hunk 3 above)
Comment (chloealee) [inline app/Http/Controllers/CampaignController.php]: :x: gone!

**Comment by jonuy** [comment]
:+1:

File: app/Services/DrupalAPI.php (hunk 4 above)
Comment (chloealee) [inline app/Services/DrupalAPI.php]: Added a new comment block - the only thing I'm missing here is the link to the documentation. I was going to wait until after it's approved to write that but should I just go ahead and write it now and add the `@see` comment line?

Comment (jonuy) [inline app/Services/DrupalAPI.php]: oh hm, sure

**Pull request merged**